### PR TITLE
Addition of generic workflows from `ACCESS-NRI/access-om2-configs`

### PR DESCRIPTION
**Pull request by CodeGat**
This PR moves the deleted workflows from https://github.com/ACCESS-NRI/access-om2-configs/pull/117/files into this repo, references the newly-moved workflows, fixes some bugs, and adds a fresh coat of paint. 

In this PR:
* Addition of `pr-*` and `schedule-*` workflows from https://github.com/ACCESS-NRI/access-om2-configs/pull/117 in https://github.com/ACCESS-NRI/model-config-tests/pull/24/commits/13cd82771a4a68e6f93397c9be4725a66ea4cac2
* Addition of `parse-ci-config` action, also in https://github.com/ACCESS-NRI/model-config-tests/pull/24/commits/13cd82771a4a68e6f93397c9be4725a66ea4cac2
* Generification of the above workflows - for those changes see https://github.com/ACCESS-NRI/model-config-tests/pull/24/files/13cd82771a4a68e6f93397c9be4725a66ea4cac2..c9d618448dc5ac7f3aca131f81c6cd416bce1d7d, but they include:
  * Replacing old `run: gh pr comment ...` commands with `uses: access-nri/actions/.github/actions/pr-comment@main` actions
  * Removing unneeded checkouts and `env:` sections
  * Renaming of `generate-initial-checksums.yml` -> `generate-checksums.yml` in https://github.com/ACCESS-NRI/model-config-tests/pull/24/commits/b87ed06167e602a6fca607acad659f1e09222a20
  * Renaming of `config-pr-checks.yml` to `test-repro.yml` in https://github.com/ACCESS-NRI/model-config-tests/pull/24/commits/9b5645b32d2e49a833c71d5b128ab137aae36db6 and removing the `model-name` input
  * Adding validation to the `config/ci.json`, just in case, in https://github.com/ACCESS-NRI/model-config-tests/pull/24/commits/1d6a1721cd0d7b47df4c37264a11ef92548e0f40
  * And fixing of a caching bug from https://github.com/ACCESS-NRI/model-config-tests/pull/22 in https://github.com/ACCESS-NRI/model-config-tests/pull/24/commits/76157f8af4b03be53ffeed1b171c4d025428fcf4
* `README.md`: new reusable workflows
* `CI.yml`: Ignore running checks on PRs that modify CI/CD and some other supplementary files

References #23

### CODE REVIEW

**Review by jo-basevi — COMMENTED** [review]
Changes look good! I've added a few minor comments. 

I might update the `how to run tests manually` part of the readme, to use virtual environments (so to not install to local directory) and to not use payu modules (as its installed as part of model-config-tests). I can create a separate PR for that section, or add a commit to this one?

File: README.md [hunk 1]
```diff
@@ -74,3 +81,48 @@ e.g.:
...
+
+Using it has some requirements outside of just filling in the inputs: One must have a valid GitHub Environment (specified by the `environment-name` input) in the calling repository, that has the following `secrets` and `vars` defined:
+
+- `secrets.SSH_HOST` - hostname for the deployment target
+- `secrets.SSH_HOST_DATA` - hostname for the data mover on the deployment target (if it exists)
+- `secrets.SSH_KEY` - private key for access to the deployment target
+- `secrets.SSH_USER` - username for access to the deployment target
+- `vars.EXPERIMENTS_LOCATION` - directory on the deployment target that will contain all the experiments used during testing of reproducibility across multiple runs of this workflow (ex. `/scratch/some/directory/experiments`)
+- `vars.PAYU_MODULE_LOCATION` - location of the Payu module on the deployment target (ex. `/g/data/vk83/modules`)
+- `vars.PAYU_VERSION` - version of Payu to use for the experiments (ex. `1.1`)
```
[the remark this answers is not in the document]
Comment (CodeGat) [reply README.md:128]: Good point, will remove

**Comment by CodeGat** [comment]
@jo-basevi feel free to commit to this one if it's related :)

File: .github/workflows/config-pr-1-ci.yml [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+    # Run the given config on the deployment GitHub Environment (`environment-name`) and
+    # upload the checksums and test details
+    needs:
+      - commit-check
+      - branch-check
+      - config
+    if: needs.commit-check.outputs.authorship != vars.GH_ACTIONS_BOT_GIT_USER_NAME && needs.branch-check.result == 'success'
+    uses: access-nri/model-config-tests/.github/workflows/test-repro.yml@main
+    with:
+      model-name: access-om2
```
[the remark this answers is not in the document]
Comment (CodeGat) [reply .github/workflows/config-pr-1-ci.yml]: Removed the input altogether as we can just infer the name from the callers `github.event.repository.name`

File: .github/workflows/config-schedule-2-start.yml [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+          config-filepath: "config/ci.json"
+
+  repro-ci:
+    # Run the given config on the deployment Github Environment (`environment-name`) and upload
+    # the test results and checksum.
+    needs:
+      - config
+    uses: access-nri/model-config-tests/.github/workflows/test-repro.yml@main
+    with:
+      model-name: access-om2
```
[the remark this answers is not in the document]
Comment (CodeGat) [reply .github/workflows/config-schedule-2-start.yml]: Similarly, removed the input altogether as we can just infer the name from the callers `github.event.repository.name`

File: .github/workflows/config-pr-1-ci.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jo-basevi) [reply .github/workflows/config-pr-1-ci.yml]: Cool!

**Review by jo-basevi — APPROVED** [review]
LGTM

**Pull request merged**